app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/catalog/ui/view/CatalogScreen.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ *
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *
+ */
+
+package com.microsoft.device.samples.dualscreenexperience.presentation.catalog.ui.view
+
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clipToBounds
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.min
+import com.microsoft.device.samples.dualscreenexperience.domain.catalog.model.CatalogItem
+import com.microsoft.device.samples.dualscreenexperience.presentation.catalog.utils.PagerState
+import com.microsoft.device.samples.dualscreenexperience.presentation.catalog.utils.ViewPager
+import com.microsoft.device.samples.dualscreenexperience.presentation.catalog.utils.rememberViewPagerState
+import kotlinx.coroutines.launch
+import kotlin.math.abs
+
+@Composable
+fun Catalog(
+    pane1WidthDp: Dp,
+    pane2WidthDp: Dp,
+    isDualScreen: Boolean,
+    foldSizeDp: Dp,
+    isFeatureHorizontal: Boolean,
+    isSinglePortrait: Boolean,
+    showTwoPages: Boolean,
```

Comment:
I don't think you need both `isDualScreen` and `showTwoPages`, since they do the same job here.

Reply:
Good observation! I will use showTwoPages because I need to have extra checkings like if the folding future is vertical & if the device has large screen or hinge.
`   val shouldShowTwoPages = windowLayoutInfo.isInDualMode() &&
            windowLayoutInfo.isFoldingFeatureVertical() &&
            isLargeScreenOrHasHinge`